hal/halx86/acpi/halpnpdd.c

```diff
@@ -153,6 +153,30 @@ HalpQueryInterface(IN PDEVICE_OBJECT DeviceObject,
                    IN PINTERFACE Interface,
                    OUT PULONG Length)
 {
+    // See /sdk/include/ddk/wdmguid.h.
+    const GUID GUID_ACPI_REGS_INTERFACE_STANDARD =
+        { 0x06141966, 0x7245, 0x6369, { 0x46, 0x2e, 0x4e, 0x65, 0x6c, 0x73, 0x6f, 0x6e } };
+    const GUID GUID_ACPI_PORT_RANGES_INTERFACE_STANDARD =
+        { 0xf14f609b, 0xcbbd, 0x4957, { 0xa6, 0x74, 0xbc, 0x00, 0x21, 0x3f, 0x1c, 0x97 } };
+
+    if (IsEqualIID(InterfaceType, &GUID_ACPI_REGS_INTERFACE_STANDARD))
+    {
+        DPRINT1("InterfaceType: %s\n", "GUID_ACPI_REGS_INTERFACE_STANDARD");
+    }
+    else if (IsEqualIID(InterfaceType, &GUID_ACPI_PORT_RANGES_INTERFACE_STANDARD))
+    {
+        DPRINT1("InterfaceType: %s\n", "GUID_ACPI_PORT_RANGES_INTERFACE_STANDARD");
+    }
+    else
+    {
+        DPRINT1("InterfaceType: {%08X-%04X-%04X-%02X%02X-%02X%02X%02X%02X%02X%02X}\n",
+                InterfaceType->Data1, InterfaceType->Data2, InterfaceType->Data3,
+                InterfaceType->Data4[0], InterfaceType->Data4[1],
+                InterfaceType->Data4[2], InterfaceType->Data4[3],
+                InterfaceType->Data4[4], InterfaceType->Data4[5],
+                InterfaceType->Data4[6], InterfaceType->Data4[7]);
+    }
+
     UNIMPLEMENTED;
```

Comment:
I would prefer replacing this rather than adding to it. More information in the logs is better, more lines, not so much.
For example the print could look like:
```
HalpQueryInterface(<guid here>) is UNIMPLEMENTED
```

Reply:
Done.